// db/dbConnect.js
var mysql = require('mysql');

function connectServer() {

    var client = mysql.createConnection({
        host: 'localhost',
        port: '3306',
        user: 'root',
        password: 'whr1997',
        database: 'User'
    });
    /*
    conn.connect(function(err) {
        if (err) {
            console.log('error when connecting to db:', err);
            setTimeout(connectServer, 2000);
        }
    });

    conn.on('error', function(err) {
        console.log('db error', err);
        // 如果是连接断开，自动重新连接
        if (err.code === 'PROTOCOL_CONNECTION_LOST') {
            connectServer();
        } else {
            throw err;
        }
    });
    */
    return client;
}

//select function
function selectFun(client, username, callback) {
    //client为一个mysql连接对象
    var sqlselect = 'select * from userinfo where username=';
    client.query(sqlselect + '"' + username + '"', function(err, results) {
        if (err) {
            console.log('[SELECT ERROR] - ', err.message);
            return err;
        }
        callback(results);
    });
}
//insert function
function insertFun(client, username, password, contact, callback) {
    client.query('insert into userinfo value(?,?,?)', [username, password, contact],
        function(err, result) {
            if (err) {
                console.log("error:" + err.message);
                return err;
            }
            callback(err);
        });
}

exports.connect = connectServer;
exports.selectFun = selectFun;
exports.insertFun = insertFun;
